Set default policy server as orfinary resource and allow users to disable it (#65)
kubewarden/helm-charts: charts/kubewarden-defaults/README.md

# kubewarden-defaults

> `kubewarden-defaults` is the Helm chart that installs a default Policy Server

Reviewer: ```suggestion
`kubewarden-defaults` is the Helm chart that installs a default PolicyServer
```

required by the Kubewarden to run `ClusterAdmissionPolicy`. It should be installed
before installing any policies.

## Installing

For example:
```console
$ helm repo add kubewarden https://charts.kubewarden.io
$ helm install --create-namespace -n kubewarden kubewarden-defaults kubewarden/kubewarden-defaults
```

For a more comprehensive documentation about how to install the whole Kubewarden
stack, check the `kubewarden-controller` chart documentation out.

## Upgrading the charts

Please refer to the release notes of each version of the helm charts.
These can be found [here](https://github.com/kubewarden/helm-charts/releases).

## Uninstalling the charts

To uninstall/delete kubewarden-crds use the following command:

```console
$ helm uninstall -n kubewarden kubewarden-defaults
```

The commands remove all the Kubernetes components associated with the chart.

> Keep in mind that the chart is required by the `kubewarden-controller` chart.

Reviewer 2: Is that correct? I think this chart can be removed without affecting the `kubewarden-controller`. The removal of the chart is however going to cause all the policies running on the `default` Policy Server to be removed. This is something that should be written in big letters, both here and inside of our docs.

# kubewarden-defaults

`kubewarden-defaults` is the Helm chart that installs a default PolicyServer
required by the Kubewarden to run `ClusterAdmissionPolicy`. It should be installed
before installing any policies.

The chart allows the user to install some default policies to enforce some
best practice security checks. By the default, the policies are disable and the
user must enables this feature. The default policies are:

- https://github.com/kubewarden/allow-privilege-escalation-psp-policy: prevents process to gain more privileges.
- https://github.com/kubewarden/host-namespaces-psp-policy: blocks pods trying to share host's IPC, networks and PID namespaces
- https://github.com/kubewarden/pod-privileged-policy: does not allow pod running in privileged mode

> - https://github.com/kubewarden/user-group-psp-policy: prevents pod running with root user

Author: > nitpicking, I would use the markdown syntax:
> 
> ```
> [allow provilege escalation policy](http://example.com): ....
> ```

Ops, it's not nitpicking. I forgot to format. This is not on purpose. xD

# kubewarden-defaults

`kubewarden-defaults` is the Helm chart that installs a default PolicyServer
required by the Kubewarden to run `ClusterAdmissionPolicy`. It should be installed
before installing any policies.

## Enable recommended policies

The chart allows the user to install some recommended policies to enforce some

> best practice security checks. By the default, the policies are disable and the

Reviewer 3: ```suggestion
best practice security checks. By the default, the policies are disabled and the
```

- [host namespaces policy](https://github.com/kubewarden/host-namespaces-psp-policy): blocks pods trying to share host's IPC, networks and PID namespaces
- [pod privileged policy](https://github.com/kubewarden/pod-privileged-policy): does not allow pod running in privileged mode
- [user-group policy](https://github.com/kubewarden/user-group-psp-policy): prevents pod running with root user

All the policies are installed cluster wide. But they are configured to ignore
namespaces important to run the control plane and Rancher components, like
`kube-system` and `rancher-operator-system` namespaces.

Furthermore, all the policies are installed in "monitor" mode by default. This
means that the policies will **not** block requests. They will report the requests
which violates the policies rules. To change the default policy mode to "protect" mode,
the user can change the default policy mode using the Helm chart value.

For example, if the user wants to install the policies in "protect" mode and ignore the
resources from the "kube-system" and "devel" namespaces, the following command can be used:

```

> helm install --set recommendedPolicies.enabled=True --set recommendedPolicies.skipNamespaces=\{kube-system,devel\} --set recommendedPolicies.defaultPolicyMode=protect kubewarnde-defaults kubewarden/kubewarden-defaults

Reviewer 3: ```suggestion
helm install --set recommendedPolicies.enabled=True --set recommendedPolicies.skipNamespaces=\{kube-system,devel\} --set recommendedPolicies.defaultPolicyMode=protect kubewarden-defaults kubewarden/kubewarden-defaults
```
